File: Python/LinkedList/Remove every k'th node.py
//https://www.geeksforgeeks.org/problems/remove-every-kth-node/1
class Solution:
    def deleteK(self, head, k):
        #code here  
        if k == 1:
            return None
        if k == 0:
            return head
        
        count = 0
        temp = head
        while temp:
            count += 1
            if (count + 1) % k == 0:
                if temp.next:
                    temp.next = temp.next.next
                count += 1
            temp = temp.next
        return head

File: Java/Linked List/Remove every k'th node.java
//https://www.geeksforgeeks.org/problems/remove-every-kth-node/1
class Solution {
    /*You are required to complete this method*/
    Node deleteK(Node head, int k) {
        // Your code here
        if (k == 1) return null; // Delete every node
        if (k == 0 || head == null) return head; // Invalid k or empty list

        Node current = head;
        int count = 1;

        while (current != null && current.next != null) {
            if ((count + 1) % k == 0) {
                // delete k-th node (which is current.next)
                current.next = current.next.next;
                count++; // Skip deleted node in count
            } else {
                current = current.next;
                count++;
            }
        }

        return head;
    }
}
